### PR TITLE
[WIP] add job_board extension [merge after conference]

### PR DESCRIPTION
**Pull request by dringler**
added job board extension to automatically post the jobs from google form csv export to the job board discord forum.

### CODE REVIEW

File: EuroPythonBot/extensions/job_board/Job Posts 2024 (Antworten) - Formularantworten 2.csv [hunk 1]
```diff
@@ -0,0 +1,294 @@
+Zeitstempel,E-Mail-Adresse,Job title,Company name,Job Location,Deadline,Job description,Duties and responsibilities,Appreciated qualification and experience,Additional info (optional),Contact person name,Contact person email,URL/Link to job posting,Picture for job posting (if available)
```
Comment (theopinard) [inline EuroPythonBot/extensions/job_board/Job Posts 2024 (Antworten) - Formularantworten 2.csv]: [minor] this file name is very German ^^'

File: posted_jobs.txt
Comment (theopinard) [inline posted_jobs.txt:1]: this should not exist, it should be only created on the server.

File: EuroPythonBot/extensions/job_board/__init__.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+@attrs.define
+class JobBoard(commands.Cog):
+    """Set up the job board extension."""
+
+    _bot: commands.Bot
+    _config: configuration.Config
+
+    @commands.Cog.listener()
+    async def on_ready(self):
```
Comment (theopinard) [inline EuroPythonBot/extensions/job_board/__init__.py:31]: not fully sure I understand when it will trigger itself?

Comment (dringler) [reply EuroPythonBot/extensions/job_board/__init__.py:31]: Same! I just reused the code from the registration.
Discord magic, see https://discordpy.readthedocs.io/en/stable/api.html#discord.on_ready

File: posted_jobs.txt
Comment (dringler) [reply posted_jobs.txt:1]: renamed the file :)

File: EuroPythonBot/extensions/job_board/Job Posts 2024 (Antworten) - Formularantworten 2.csv (hunk 1 above)
Comment (dringler) [reply EuroPythonBot/extensions/job_board/Job Posts 2024 (Antworten) - Formularantworten 2.csv]: renamed the file :)

File: posted_jobs.txt
Comment (dringler) [reply posted_jobs.txt:1]: deleted the file

**Review by theopinard — APPROVED** [review]

**Pull request merged**